### PR TITLE
CORE-760: Support API_ONLY mode for BCH

### PR DESCRIPTION
**Pull request by EBGToo**

### CODE REVIEW

File: bitcoin/BRSyncManager.c [hunk 1]
```diff
@@ -1211,6 +1214,44 @@ BRClientSyncManagerAnnounceGetTransactionsItem (BRClientSyncManager manager,
     }
 }
 
+static BRArrayOf(char *)
+BRClientSyncManagerConvertAddressToString (BRClientSyncManager manager,
+                                           OwnershipGiven BRArrayOf(BRAddress *) addresses) {
+    if (NULL == addresses || 0 == array_count(addresses))
```
Comment (michael-brd) [inline bitcoin/BRSyncManager.c]: leak if count is 0?

Comment (EBGToo) [reply bitcoin/BRSyncManager.c]: Yes.

File: bitcoin/BRSyncManager.c [hunk 2]
```diff
@@ -2041,8 +2087,14 @@ static BRAddress *
 _getWalletAddresses (BRWallet *wallet,size_t *addressCount) {
     assert (addressCount);
 
+    // For BTC, `BRWalletAllAddrs() produces an array of SEGWIT addresses; for BCH, it
+    // produces an array of LEGACY addresses.  See `BRAddressFromHash160()`
     size_t addrCount = BRWalletAllAddrs (wallet, NULL, 0);
 
+    // We'll double up the number of addresses.  For BTC: SEGWIT + LEGACY; for BCH: LEGACY +
```
Comment (michael-brd) [inline bitcoin/BRSyncManager.c]: Is it too much trouble to propagate if it is BCH/BTC to this point?

Comment (EBGToo) [reply bitcoin/BRSyncManager.c]: I thought about it but biased towards minimal changes - once I realized BRSet would handle it.  Better to just pass an argument through?

Comment (EBGToo) [reply bitcoin/BRSyncManager.c]: Maybe better to handle it a different way...  Don't add the LEGACY address here; instead when converting to string, for BTC, double up with SEGWIT + LEGACY? That then leaves BRWalletAllAddrs() as the set of addresses and leaves the string conversion for BTC and BCH in one place.

Comment (michael-brd) [reply bitcoin/BRSyncManager.c]: I like that.

**Pull request merged**